### PR TITLE
[WiP] Drupal 10 example

### PR DESCRIPTION
**Pull request by back-2-95**
- Worked mode with `el7cosmos/drupal-runtime`
- Drupal 10 (it will be released around same time as 9.5)
- Alpine variants for FrankenPHP and PostgreSQL
- Add `apcu` extension as Drupal recommends it
- Install `pg_trgm` extension to PostgreSQL image
- Copy Composer from slim `composer/composer:2-bin` image for faster build
- User `composer.json` as configurable start point for Drupal installation
- Use `/app/public` as webroot

## Notes

- Drupal 9.5 gave a lot of warnings throughout the install process. Drupal 10 gave none 💪 
- Worker mode acts a bit weird where pages not load until 2nd or 3rd click/refresh

### CODE REVIEW

**Comment by back-2-95** [comment]
`composer.json` has `drupal/core-composer-scaffold` and `composer/installers` which allow as to set webroot as `public` among other things.

**Comment by back-2-95** [comment]
Drush 11 causes error so not added here.

**Comment by back-2-95** [comment]
PostgreSQL extension `pg_trgm` is hard requirement for Drupal 10.

**Comment by back-2-95** [comment]
Drupal 9 has Symfony 4 and Drupal 10 has Symfony 6

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/.idea/
```
Comment (dunglas) [inline .gitignore]: Can this be removed please?

File: docker-compose.yml [hunk 2]
```diff
@@ -27,9 +26,34 @@ services:
       # existing content of the image at the same location
       - /app/public/sites
     restart: always
+    environment:
+      FRANKENPHP_CONFIG: "worker ./public/index.php"
+      SERVER_NAME: :80
```
Comment (dunglas) [inline docker-compose.yml:31]: Can you remove these labels please? We try to keep these example repositories as minimal as possible.

File: docker-entrypoint-initdb.d/init-pg_trgm-extension.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -e
+
+psql -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" --dbname "$POSTGRES_DB" <<-EOSQL
+	CREATE EXTENSION pg_trgm;
```
Comment (dunglas) [inline docker-entrypoint-initdb.d/init-pg_trgm-extension.sh:6]: They don't do that in the official image. Why is it necessary?

File: docker-compose.yml [hunk 4]
```diff
@@ -27,9 +26,34 @@ services:
...
+    container_name: db
+    build:
+      context: .
+      dockerfile: db.Dockerfile
+    image: frankenphp-drupal/postgres
+    volumes:
+      - /var/lib/postgresql/data
     environment:
       POSTGRES_PASSWORD: example
     restart: always
```
Comment (dunglas) [inline docker-compose.yml:43]: Same here, can we remove this?

File: docker-compose.yml (hunk 2 above)
Comment (back-2-95) [reply docker-compose.yml:31]: I'll move them to another branch on our fork.

File: .gitignore (hunk 1 above)
Comment (back-2-95) [reply .gitignore]: I'll move them to another branch on our fork.

File: docker-compose.yml (hunk 4 above)
Comment (back-2-95) [reply docker-compose.yml:43]: I'll move them to another branch on our fork.

File: docker-entrypoint-initdb.d/init-pg_trgm-extension.sh (hunk 3 above)
Comment (back-2-95) [reply docker-entrypoint-initdb.d/init-pg_trgm-extension.sh:6]: Drupal 10 has hard requirement on that extension and it was documented that you "enable" it this way. As it's not a package.

- https://www.drupal.org/node/3214907
- https://www.drupal.org/project/farm/issues/3270558
- https://gist.github.com/leopoldodonnell/b0b7e06943bd389560184d948bdc2d5b

**Comment by dunglas** [comment]
Could we try to use the official installer and the official Docker image as much as possible, and to stay as close as possible from a default Drupal directory structure?

IMHO we should make the experience as seamless as possible for existing Drupal devs. So changing FrankenPHP config to match Drupal's expectation is better than changing Drupal config or structure.

**Comment by back-2-95** [comment]
I'll move this to issues and just link to fork where it's more like alternative setup.

**Comment by dunglas** [comment]
Having support for Drupal 10 with worker mode on would be great however.

**Comment by back-2-95** [comment]
For going Drupal 10 maybe we could consider these 2 options for the database:
- change to mariadb/mysql container as it's more used in Drupal scene than Postgres
- if using Postgres, we need to use Dockerfile to install that required `pg_trgm` extension

**Pull request closed without merging**